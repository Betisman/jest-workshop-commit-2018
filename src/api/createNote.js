
export default async (title, description) => {
  const response = await fetch('http://localhost:3001/note', {
    method: 'post',
    body: JSON.stringify({ title, description }),
    headers: { 'Content-Type': 'application/json' },
  });
  const notes = await response.json();
  return notes;
};