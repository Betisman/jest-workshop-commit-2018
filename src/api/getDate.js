
export default async () => {
  const response = await fetch('http://worldclockapi.com/api/json/est/now');
  const date = await response.json();
  return date;
};
