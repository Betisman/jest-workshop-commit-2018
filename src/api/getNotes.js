
export default async () => {
  const response = await fetch('http://localhost:3001/notes');
  const notes = await response.json();
  return notes;
};
