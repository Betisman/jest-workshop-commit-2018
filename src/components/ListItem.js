import React from 'react';

class ListItem extends React.Component {
  constructor() {
    super();
    this.state = { open: false };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({ open: !this.state.open });
  }

  render() {
    const { open } = this.state;
    const { title, description } = this.props;
    return (
      <div className="card">
        <div className="content">
          <div className="header" onClick={this.handleClick}>{title}</div>
          {open && (
            <div className="description">{description}</div>
          )}
        </div>
      </div>
    );
  }
}

export default ListItem;
