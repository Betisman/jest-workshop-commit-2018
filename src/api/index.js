import getDate from './getDate';
import createNote from './createNote';
import getNotes from './getNotes';

export {
  getDate,
  createNote,
  getNotes,
};
