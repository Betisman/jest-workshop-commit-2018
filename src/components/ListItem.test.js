import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ListItem from './ListItem';
import { note } from '../../test/fixtures/api';

const props = {
  ...note,
};

describe('<ListItem />', function() {
  it('Match snapshot', () => {
    const tree = renderer.create(<ListItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not show description value', () => {
    const wrapper = shallow(<ListItem {...props} />);
    expect(wrapper.find('.description').exists()).toEqual(false);
  });

  it('should show description value after header click', () => {
    const wrapper = shallow(<ListItem {...props} />);
    wrapper.find('.header').simulate('click');
    expect(wrapper.find('.description').exists()).toEqual(true);
  });
});