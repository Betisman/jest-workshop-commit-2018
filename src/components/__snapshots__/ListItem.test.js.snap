// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> Match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="content"
  >
    <div
      className="header"
      onClick={[Function]}
    />
  </div>
</div>
`;
